src/uu/seq/src/floatparse.rs

```diff
@@ -0,0 +1,230 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore extendedbigdecimal bigdecimal hexdigit numberparse
+use crate::extendedbigdecimal::ExtendedBigDecimal;
+use crate::number::PreciseNumber;
+use crate::numberparse::ParseNumberError;
+use bigdecimal::BigDecimal;
+use num_traits::FromPrimitive;
+
+///  Parse a number from a floating-point hexadecimal exponent notation.
+///
+/// # Errors
+///
+/// This function returns an error if:
+/// - the input string is not a valid hexadecimal string
+/// - the input data can't be interpreted as ['f64'] or ['BigDecimal']
+///
+/// # Examples
+///
+/// ```rust,ignore
+/// let input = "0x1.4p-2";
+/// let expected = 0.3125;
+/// match input.parse::<PreciseNumber>().unwrap().number {
+///     ExtendedBigDecimal::BigDecimal(bd)  => assert_eq!(bd.to_f64().unwrap(),expected),
+///     _ => unreachable!()
+/// };
+/// ```
+pub fn parse_hexadecimal_float(s: &str) -> Result<PreciseNumber, ParseNumberError> {
+    let value = parse_float(s)?;
+    let number = BigDecimal::from_f64(value).ok_or(ParseNumberError::Float)?;
```

Comment:
in case of floating points, seq already supports arbitrary precision floating points, i.e.:
`uu_seq 1e9865 1e9864 2e9865`
the above limits hexadecimal support to f64, so `p1023` is maximal binary exponent supported.

(not sure if that is an issue or not, but imo it would be nice to have arbitrary precision supported in both variants)

Reply:
@gim913 Thanks, this is a good point. I also thought about it.  And:
1. I decided to start with native f64 just to have something to demonstrate and discuss. :) 
2. arbitrary precision is a great option, but we should also consider compatibility with the original seq. From what I can see, GNU seq operates on ```long double```'s with a ```15-bit exponent```. This should serve as a bound for arbitrary precision. For example,
```
seq 0x1.4p16383 0
seq 0x1.4p16384 0
seq: invalid floating point argument: ‘0x1.4p16384’
Try 'seq --help' for more information.
```